### PR TITLE
Clock offset measurement

### PR DESCRIPTION
**Pull request by darrylmasson**
A bit crude, but the difference between the `_id` and `time` fields in the system monitor collection is a way of monitoring the system clock. This PR adds this quantity to the Hosts page.

### CODE REVIEW

File: public/javascripts/hosts.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ function UpdateMonitorPage(){
       $(`#${h}_mem_used`).html((data['virtual_memory']['percent']).toFixed(1)+"%").css("color", data['virtual_memory']['percent'] > 75 ? 'red' : 'black');
       $(`#${h}_swap`).html((data['swap_memory']['percent']).toFixed(1) + "%").css("color", data['swap_memory']['percent'] > 75 ? 'red' : 'black');
       $(`#${h}_load`).html(data['load'].toString());
+      $(`#${h}_drift`).html(data['drift'].toFixed(1) + ' s');
```
Comment (JoranAngevaare) [inline public/javascripts/hosts.js:28]:
```suggestion
      $(`#${h}_drift`).html(data['drift'].toFixed(1) + ' s').css("color", Math.abs(data['drift']) > 2 ? 'red' : 'black');
```
Not sure how java does/needs imports

**Review by JoranAngevaare — APPROVED** [review]
Neat idea, I like it.

Just for my understanding the `'_id'` field gets the timestamp from the host of the database (in our case `gw`)?

This pr makes me wonder if we could not just remove all the `time` fields and just use the `_id` field of documents in `redax'` logic. Probably not at all worth the effort though.

**Comment by darrylmasson** [comment]
The `_id` contains the timestamp of the server hosting the database, and `time` is the timestamp of the server making the insert. We can't get rid of `time` because that's the TTL index that keeps the collections to a manageable size, and it's easier than running a cronjob. I also don't know how to automatically add fields like explicit timestamps on insert, only on update, and running everything as an upsert is inefficient and inelegant. All the important logic in the dispatcher uses gw's timestamps, and as long as oldmaster doesn't drift more than 5 seconds we shouldn't fall afoul of the 23.12 issue. If we're sufficiently concerned I can add a few lines that force the dispatcher to use gw's clock for everything and not rely on oldmaster's clock.

Also this PR is currently running on /test/, so you can see what it looks like.

**Comment by JoranAngevaare** [comment]
thanks 👍  I like it a lot

**Pull request merged**